server/hotrod/src/main/scala/org/infinispan/server/hotrod/iteration/IterationManager.scala

```diff
@@ -80,7 +80,7 @@ class DefaultIterationManager(val cacheManager: EmbeddedCacheManager) extends It
 
    override def start(cacheName: String, segments: Option[JavaBitSet], filterConverterFactory: Option[String], batch: Integer): IterationId = {
       val iterationId = UUID.randomUUID().toString
-      val stream = cacheManager.getCache(cacheName).getAdvancedCache.cacheEntrySet.stream().asInstanceOf[CacheStream[_]]
+      val stream = cacheManager.getCache(cacheName).getAdvancedCache.cacheEntrySet.stream().asInstanceOf[CacheStream[CacheEntry[AnyRef, AnyRef]]]
```

Comment:
> I had a look at this and dunno how to avoid the cast

Me neither. After some debugging, I saw that the error occurs very early in the ```typer``` compiler phase

Anyway, created https://issues.scala-lang.org/browse/SI-9432